### ISSUE TITLE
zphysics does not compile and support MSVC ABI

### ISSUE THREAD

**Issue by mlarouche** [michal-z/zig-gamedev#290]
While building a Zig binding to a C++ API myself (not public yet), I am to the point where I need to bind C++ class with virtual methods. Contrary to Jolt, my C++ API is closed source and built with MSVC. 

So when trying to replicate what you did, I ran into issues because the vtable layout is different when compiling with MSVC ABI. For now, it is not really an issue for you, but it will come a time that a user will try to use `zphysics` with the MSVC ABI for a game/project that want to be compatible with closed-source libraries (like FMOD, Steam, Wwise) and will not compile and works even after the compile errors are fixed.

So this is what I have done on my side:

In the C header file, I create a macro that will expand the vtable header depending if we are compiling with MSVC ABI or not.
```cpp
#if defined(_MSC_VER)
#define MYLIBC_VTABLE_HEADER const void* __vtable_header[1]
#else
#define MYLIBC_VTABLE_HEADER const void* __vtable_header[2]
#endif

// [...]
typedef struct MYLIBC_TestVTable
{
   MYLIBC_VTABLE_HEADER;

   void (*Print)(void* in_self, int value);
} MYLIBC_TestVTable;
```

In Zig I have make the following type to handle the header:
```zig
pub const VTableHeader = switch (builtin.abi) {
    .msvc => extern struct {
        __header: ?*const anyopaque = null,
    },
    else => extern struct {
        __header: [2]?*const anyopaque = [_]?*const anyopaque{null} ** 2,
    },
};
```

that gets used like this:
```zig
pub const VTable = extern struct {
        __vtable_header: common.VTableHeader = .{},
        print: *const fn (self: *TestVTable, value: i32) callconv(.C) void,
};
```

**Comment by michal-z** [comment]
Thanks for the report!

Will fix `zphysics` next week.

**Comment by michal-z** [comment]
I get compile errors in `libcxx` and `libcxxabi` when using `msvc` ABI with latest zig compiler. Is this a zig compiler issue?

**Comment by mlarouche** [comment]
I was not able to find a quick workaround for compile errors sadly :(

**Comment by michal-z** [comment]
How do you use `msvc` ABI then? You don't get those errors when compiling your project?

**Comment by mlarouche** [comment]
I'm implementing a binding to different C++ API, and I was just using your zphysics implementation as a example for my binding

**Comment by michal-z** [comment]
Yes I understand, so those errors are Jolt/zphysics specific?

**Comment by mlarouche** [comment]
The virtual table layout issue is not specific to Jolt/zphysics.

**Comment by michal-z** [comment]
Yes, sorry I wasn't clear enough. I see the problem with virtual tables and I want to fix it.

But, when trying to build `Jolt` with `msvc` ABI I get compile errors in code that is shipped with Zig compiler (`libcxx`, `libcxxabi`).

Thanks! I will keep digging.

**Comment by michal-z** [comment]
All `zphysics` tests pass now with both `msvc` and `gnu` ABIs.
